File: escomvoting/src/model/response/UrnResponse.ts
import type { BallotDTO } from './BallotDTO'
import type { CandidateDTO } from './CandidateDTO'
import type { PageResponse } from './PageResponse'

export interface UrnResponse {
  electionId: string
  electionTitle: string
  status: string
  candidates: CandidateDTO[]
  ballots: PageResponse<BallotDTO>
}

File: escomvoting-backend/src/main/java/com/github/escom/escomvoting/model/dto/UrnResponse.java
package com.github.escom.escomvoting.model.dto;

import java.util.List;
import java.util.UUID;

public record UrnResponse(
        UUID electionId,
        String electionTitle,
        List<CandidateDTO> candidates,
        PageResponse<BallotDTO> ballots
) {}
